pkg/template/controller/templateinstance_controller.go

```diff
@@ -233,18 +233,22 @@ func (c *TemplateInstanceController) instantiate(templateInstance *templateapi.T
 
 	u := &user.DefaultInfo{Name: templateInstance.Spec.Requester.Username}
 
-	if err := util.Authorize(c.kc.Authorization().SubjectAccessReviews(), u, &authorization.ResourceAttributes{
-		Namespace: templateInstance.Namespace,
-		Verb:      "get",
-		Group:     kapi.GroupName,
-		Resource:  "secrets",
-	}); err != nil {
-		return err
-	}
+	var secret *kapi.Secret
+	if templateInstance.Spec.Secret != nil {
+		if err := util.Authorize(c.kc.Authorization().SubjectAccessReviews(), u, &authorization.ResourceAttributes{
+			Namespace: templateInstance.Namespace,
+			Verb:      "get",
+			Group:     kapi.GroupName,
+			Resource:  "secrets",
+		}); err != nil {
+			return err
+		}
 
-	secret, err := c.kc.Core().Secrets(templateInstance.Namespace).Get(templateInstance.Spec.Secret.Name, metav1.GetOptions{})
-	if err != nil {
-		return err
+		s, err := c.kc.Core().Secrets(templateInstance.Namespace).Get(templateInstance.Spec.Secret.Name, metav1.GetOptions{})
```

Comment:
no, because that shadows the secret var defined above (which must be defined above because it needs to exist outside this scope).  and i can't use "=" because err is actually a new variable.

unless i'm missing a trick, this is the joys of golang combining implicit variable declaration, multivariable returns, and zero shadowing protection.